docs/workflow/testing/libraries/testing-apple.md

```diff
@@ -1,34 +1,89 @@
-# Testing Libraries on iOS and tvOS
+# Testing Libraries on iOS, tvOS, and MacCatalyst
 
-In order to build libraries and tests for iOS or tvOS you need recent version of XCode installed (e.g. 11.3 or higher).
+## Prerequisites
 
-Build Libraries for iOS Simulator:
+- XCode 11.3 or higher
+
+## Building Libs and Tests
+
+You can build and run the library tests:
+- on a simulator;
+- on a device.
+
+Run the following command in a terminal:
+```
+./build.sh mono+libs -os <TARGET_OS> -arch x64
+```
+where `<TARGET_OS>` is one of the following:
+
+| simulator     | device|
+|:--------------|:------|
+| iOSSimulator  | iOS   |
+| tvOSSimulator | tvOS  |
+| MacCatalyst   |       |
+
+
+e.g., to build for a iOS simulator, run:
 ```
 ./build.sh mono+libs -os iOSSimulator -arch x64
 ```
+
 Run tests one by one for each test suite on a simulator:
 ```
 ./build.sh libs.tests -os iOSSimulator -arch x64 -test
 ```
+
+### Building for a device
+
 In order to run the tests on a device:
-- Set the os to `iOS` instead of `iOSSimulator`
-- Specify `DevTeamProvisioning` (see [developer.apple.com/account/#/membership](https://developer.apple.com/account/#/membership), scroll down to `Team ID`):
+- Set the `-os` parameter to a device-related value (see above)
+- Specify `DevTeamProvisioning` (see [developer.apple.com/account/#/membership](https://developer.apple.com/account/#/membership), scroll down to `Team ID`).
+
+For example:
 ```
 ./build.sh libs.tests -os iOS -arch x64 -test /p:DevTeamProvisioning=H1A2B3C4D5
 ```
-[AppleAppBuilder](https://github.com/dotnet/runtime/blob/main/src/mono/msbuild/AppleAppBuilder/AppleAppBuilder.cs) generates temp Xcode projects you can manually open and resolve provisioning issues there using native UI and deploy to your devices.
+[AppleAppBuilder](https://github.com/dotnet/runtime/blob/main/src/tasks/AppleAppBuilder/AppleAppBuilder.cs) generates temp Xcode projects you can manually open and resolve provisioning issues there using native UI and deploy to your devices.
 
 ### Running individual test suites
 - The following shows how to run tests for a specific library:
```

Comment:
We can also add some details of running the test app through Xcode by opening the corresponding `.xcodeproj` and setting up the right scheme, app, and even signing if using a local device.

There's also an option to run a `.app` through xcrun, which is simulator specific. We used to use that CLI command to run the sample, documented in https://github.com/dotnet/samples/blob/153183a79fa6f1943cd99988ed39a736ff809c05/core/mono-samples/iOS/iOSSampleApp.csproj#L53-L59.

Reply:
I think we can also add documentation on using the Console app on macOS to view the logs (command + space, type in Console), and then we can search the appropriate process (System.Buffers.Tests) for example. This is another way to capture logs as in the past theres been some hiccups where the logs would not be printed properly.